darwin/DarwinProcess.c

```diff
@@ -496,7 +496,9 @@ void DarwinProcess_scanThreads(DarwinProcess* dp, DarwinProcessTable* dpt) {
          continue;
       }
 
-      assert(Process_getPid(tprocess) == tid);
+      pid_t tprocessPid = Process_getPid(tprocess);
+      assert(tprocessPid >= 0);
```

Comment:
The fix doesn't look correct. Why allow PID 0 here?
Besides, are we stuck with requiring an unsigned type for the PID argument in `ProcessTable_getProcess`?

Reply:
I understand that it may seem odd but for simplicity shouldn't this check only make sure that the pid is a non-error code? any weird edge cases should be handled by checking that pid and tid are actually the same

As for the unsigned return of `ProcessTable_getPid`, why should it be signed if it will never return a negative number? if anything it could only create more expectations in code that getPid will return a wider range of values than it actually does, leading to dead checks in the code

Reply:
> any weird edge cases should be handled by checking that pid and tid are actually the same

My concern is that `tid` might be a negative number wrongly converted to unsigned in the first place.

> As for the unsigned return of `ProcessTable_getPid`, why should it be signed if it will never return a negative number?

For consistency of use of `pid_t`. And we cannot assume `sizeof(pid_t)` equals `sizeof(uint64_t)`. Using wrong size for PID values can hide potential bugs.

Reply:
looking through the code, tid is ultimately defined by `thread_info` (mach lib function), the choice to be a uint64_t is a choice by the mach devs defining the members of the `thread_identifier_info_data_t` struct rather than a misuse on this end of the codebase - formatting this particular value into a pid_t seems possible but a wrapper function would be needed for that library call and it seems that it would be a bit "messy" (would you return a new struct that has a single member changed, or would you "translate" every `uint64_t` pid into a `pid_t` by calling a method individually when you get a `uint64_t` pid?

IMO a couple of asserts seems the best and simplest fix at the moment but if anyone was planning on overhauling that would be interesting to look at